syntax = "proto3";

package eadp.cadie.ingressserver.v1;

import "grpc/ingress_server/upsert_content.proto";
import "grpc/ingress_server/delete_content.proto";
import "grpc/ingress_server/create_assets.proto";
import "grpc/ingress_server/add_download_counts.proto";
import "grpc/ingress_server/add_ratings.proto";

option java_multiple_files = true;
option java_package = "eadp.cadie.ingressserver.v1";
option java_outer_classname = "IngressServerProto";

service IngressServer {
  rpc UpsertContent(UpsertContentRequest) returns (UpsertContentResponse);
  rpc DeleteContent(DeleteContentRequest) returns (DeleteContentResponse);
  rpc BatchCreateAssets(BatchCreateAssetsRequest) returns (BatchCreateAssetsResponse);
  rpc BatchAddDownloadCounts(BatchAddDownloadCountsRequest) returns (BatchAddDownloadCountsResponse);
  rpc BatchAddRatings(BatchAddRatingsRequest) returns (BatchAddRatingsResponse);
}
